src/pathfinder/yapf/yapf_road.cpp

```diff
@@ -75,11 +75,11 @@ class CYapfCostRoadT
 						/* Increase the cost for drive-through road stops */
 						cost += Yapf().PfGetSettings().road_stop_penalty;
 						DiagDirection dir = TrackdirToExitdir(trackdir);
-						if (!RoadStop::IsDriveThroughRoadStopContinuation(tile, tile - TileOffsByDiagDir(dir))) {
-							/* When we're the first road stop in a 'queue' of them we increase
-							 * cost based on the fill percentage of the whole queue. */
+						if (Yapf().PfDetectDestinationTile(tile, trackdir)){
```

Comment:
I modified this if statement so that `GetOccupied` would only be called on destination tiles.
In my testing I did not find calling it for every RoadStop along the route to add value because the cost added on line 76 is already very high and enough of a penalty.